client/ayon_core/pipeline/anatomy/roots.py

```diff
@@ -23,16 +23,19 @@ class RootItem(FormatObject):
     def __init__(self, parent, root_raw_data, name):
         super(RootItem, self).__init__()
         self._log = None
-        lowered_platform_keys = {}
-        for key, value in root_raw_data.items():
-            lowered_platform_keys[key.lower()] = value
+        lowered_platform_keys = {
+            key.lower(): value
+            for key, value in root_raw_data.items()
+        }
         self.raw_data = lowered_platform_keys
         self.cleaned_data = self._clean_roots(lowered_platform_keys)
         self.name = name
         self.parent = parent
 
         self.available_platforms = set(lowered_platform_keys.keys())
-        self.value = lowered_platform_keys.get(platform.system().lower())
+        self.value = lowered_platform_keys[platform.system().lower()].format(
+            **os.environ
+        )
```

Comment:
I think `format_map` (added in Py3.2) should offer better performance.
```suggestion
        self.value = lowered_platform_keys[platform.system().lower()].format_map(
            os.environ
        )
```

By the way, also note the example in the Python docs: https://docs.python.org/3/library/stdtypes.html#str.format_map

Interesting approach with using a custom class with missing there.

Reply:
Let's give sidefx a moment to test it for us... 👍